### PR TITLE
refactor: convert Builds to TS

### PR DESCRIPTION
**Pull request by vladfrangu**
Depends on #191

### CODE REVIEW

File: src/resource_clients/actor.ts [hunk 1]
```diff
@@ -1,72 +1,62 @@
...
-    /**
-     * @param {ApiClientOptions} options
-     */
-    constructor(options) {
+export class ActorClient extends ResourceClient {
+    constructor(options: ApiClientOptions) {
         super({
-            resourcePath: 'acts',
             ...options,
+            resourcePath: 'acts',
```
Comment (szmarczak) [inline src/resource_clients/actor.ts:28]: Why the change?

File: src/resource_clients/actor_collection.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import { ActorVersion } from './actor_version';
+
+/**
+ * @hideconstructor
+ */
+export class ActorCollectionClient extends ResourceCollectionClient {
+    constructor(options: ApiClientOptions) {
+        super({
+            ...options,
+            resourcePath: 'acts',
```
Comment (szmarczak) [inline src/resource_clients/actor_collection.ts:15]: same here

**Review by szmarczak — APPROVED** [review]

File: src/resource_clients/actor.ts (hunk 1 above)
Comment (vladfrangu) [reply src/resource_clients/actor.ts:28]: Wrong PR to leave a review about this, but because of this.. I will need to check if the type maybe is wrong and I should use the subresource one instead
image

Comment (vladfrangu) [reply src/resource_clients/actor.ts:28]: And the answer is yes, I should be using the other type.. I'll fix in that PR, and rebase appropriately

Comment (B4nan) [reply src/resource_clients/actor.ts:28]: already merged the other one, so just fix it in this one pls

**Review by B4nan — APPROVED** [review]
